persistence/upgrades/ravendb-5to6.md

```diff
@@ -0,0 +1,174 @@
+---
+title: RavenDB Persistence Upgrade from 5 to 6
+summary: Instructions on how to upgrade NServiceBus.RavenDB 5 to 6
+component: Raven
+related:
+ - nservicebus/upgrades/6to7
+reviewed: 2019-05-30
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 7
+---
+
+This update supports upgrading to RavenDB 4.2 while using NServiceBus 7.x. The upgrade to RavenDB 4 is a **major** upgrade. It's not backward compatible, and the [API has a lot of breaking changes](https://ravendb.net/docs/article-page/4.2/csharp/migration/client-api/introduction). Additionally, the data storage format has changed, so databases need to have their [data migrated to a new RavenDB 4 database](https://ravendb.net/docs/article-page/4.2/csharp/migration/server/data-migration).
+
+## Required RavenDB client
+
+RavenDB Persistence requires [RavenDB.Client 4.2.0](https://www.nuget.org/packages/RavenDB.Client/4.2.0) or later.
```

Comment:
Do we want to comment why are we starting support from 4.2.0 version? Some people might ask that question.

Given that we don't plan on supporting multi-master clusters I don't think we have a one?

Reply:
Well, we _could_ in a minor without then requiring a server/client upgrade as well. Also you can go straight from 3.5 to 4.2 so....why wouldn't you?

But really it's mostly because that's what we started with. What do you think?

Reply:
Yesterday I had a look at this page: https://ravendb.net/support and I learned that 4.2 is the only version with LTS. 4.0 is already expired, 4.1 will expire in 5 months. So I am retracting my question, it seems like 4.2 is the first stable and supported version from the 4.X familty.